src/server/svr_jobfunc.c

```diff
@@ -5671,3 +5671,28 @@ recreate_exec_vnode(job *pjob, char *vnodelist, char *keep_select, char *err_msg
 
 	return (rc);
 }
+
+/**
+ * @brief
+ *  action_max_run_subjobs This is action function for max_run_subjobs attribute.
+ *			   It verifies that the attribute is being set only on array jobs.
+ *
+ * @param[in]	pattr	-	attribute structure
+ * @param[in]	pobject	-	job object
+ * @param[in]	actmode	-	action mode
+ */
+int
+action_max_run_subjobs(attribute *pattr, void *pobject, int actmode)
+{
+	job * pjob = (job*)pobject;
+	int jtype;
+
+	if (pjob == NULL)
+		return PBSE_INTERNAL;
+
+	jtype = is_job_array(pjob->ji_qs.ji_jobid);
```

Comment:
Don't you need to check for NEW/ALTER?

Reply:
It does not matter, action function must always reject if the attribute is being set on a normal job.

Reply:
What about job recovery?  What happens if we reject there?  Does the job get deleted?